src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/client/debug/MixinDebugScreenOverlay.java

```diff
@@ -0,0 +1,57 @@
+package io.github.opencubicchunks.cubicchunks.mixin.core.client.debug;
+
+import java.util.List;
+
+import io.github.opencubicchunks.cubicchunks.chunk.LightHeightmapGetter;
+import io.github.opencubicchunks.cubicchunks.chunk.heightmap.LightSurfaceTrackerWrapper;
+import io.github.opencubicchunks.cubicchunks.server.CubicLevelHeightAccessor;
+import it.unimi.dsi.fastutil.longs.LongSet;
+import net.minecraft.client.gui.components.DebugScreenOverlay;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LightLayer;
+import net.minecraft.world.level.chunk.LevelChunk;
+import net.minecraft.world.level.lighting.LevelLightEngine;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(DebugScreenOverlay.class)
+public abstract class MixinDebugScreenOverlay {
+    @Shadow abstract LevelChunk getServerChunk();
+
+    @Inject(method = "getGameInformation",
+            at = @At(value = "INVOKE",
+                    target = "Lnet/minecraft/client/multiplayer/ClientLevel;getBrightness(Lnet/minecraft/world/level/LightLayer;Lnet/minecraft/core/BlockPos;)I",
+                    ordinal = 0),
+            locals = LocalCapture.CAPTURE_FAILHARD)
+    private void onGetGameInformation(CallbackInfoReturnable<List<String>> cir, String string2, BlockPos pos, Entity entity, Direction direction,
+                                      String string7, Level level, LongSet longSet, List<String> list, LevelChunk clientChunk, int i) {
+        LevelChunk serverChunk = this.getServerChunk();
```

Comment:
We still need to add proper integrated server checks here, since getServerChunk() can also return the client chunk for some reason (thanks Mojang)

Reply:
Added if checks for the integrated server prior to loading the server side data in the debug overlay